src/components.rs

```diff
@@ -356,6 +358,69 @@ pub trait EventLike: Component {
         self.property_value("LOCATION")
     }
 
+    /// Set recurrence rules
+    #[cfg(feature = "recurrence")]
+    fn recurrence(&mut self, rruleset: RRuleSet) -> &mut Self {
+        let rrules = rruleset
+            .get_rrule()
+            .iter()
+            .map(ToString::to_string)
+            .collect::<Vec<_>>()
+            .join("\n");
+
+        let rdates = rruleset
+            .get_rdate()
+            .iter()
+            .map(|dt| dt.format("%Y%m%dT%H%M%SZ").to_string())
+            .collect::<Vec<_>>()
+            .join(",");
+
+        let exdates = rruleset
+            .get_exdate()
+            .iter()
+            .map(|dt| dt.format("%Y%m%dT%H%M%SZ").to_string())
+            .collect::<Vec<_>>()
+            .join(",");
+
+        self.add_property("RRULE", rrules)
+            .add_multi_property("RDATE", &rdates)
+            .add_multi_property("EXDATE", &exdates)
+    }
+
+    /// Get recurrence rules
+    #[cfg(feature = "recurrence")]
+    fn get_recurrence(&self) -> Option<RRuleSet> {
+        let dt_start_str = self.property_value("DTSTART")?;
+        let rrule_str = self.property_value("RRULE")?;
+
+        let mut rdates_str = self
+            .multi_properties()
+            .get("RDATE")
+            .unwrap_or(&vec![])
+            .iter()
+            .map(Property::value)
+            .collect::<Vec<_>>()
+            .join(",");
+        if !rdates_str.is_empty() {
+            rdates_str = format!("\nRDATE:{rdates_str}");
+        }
+
+        let mut exdates_str = self
+            .multi_properties()
+            .get("EXDATE")
+            .unwrap_or(&vec![])
+            .iter()
+            .map(Property::value)
+            .collect::<Vec<_>>()
+            .join(",");
+        if !exdates_str.is_empty() {
+            exdates_str = format!("\nEXDATE:{exdates_str}");
+        }
+
+        let rrules = format!("DTSTART:{dt_start_str}\nRRULE:{rrule_str}{rdates_str}{exdates_str}");
+        rrules.parse::<RRuleSet>().ok()
```

Comment:
It may be better to return the error (or wrap it), so that the caller can check whether there was an error parsing the `RRULE`/there is no `RRULE` or `DTSTART` was missing.